syncopy/tests/test_connectivity.py

```diff
@@ -0,0 +1,584 @@
+
+        call = lambda cfg: connectivity(self.data, cfg)
+        run_cfg_test(call, method='corr', positivity=False)
+
+    @skip_without_acme
+    @skip_low_mem
+    def test_corr_parallel(self, testcluster=None):
+
+        ppl.ioff()
+        client = dd.Client(testcluster)
+        all_tests = [attr for attr in self.__dir__()
+                     if (inspect.ismethod(getattr(self, attr)) and 'parallel' not in attr)]
+
+        for test in all_tests:
+            test_method = getattr(self, test)
+            test_method()
+        client.close()
+        ppl.ion()
+
+
+# -- helper functions --
+
+
+def run_padding_test(call, pad_length):
+    """
+    The callable should test a solution and support
+    a single keyword argument `pad_to_length`
+    """
+
+    pad_options = [pad_length, 'nextpow2', None]
+    for pad in pad_options:
+        call(pad_to_length=pad)
+
+    # test invalid pads
+    try:
+        call(pad_to_length=2)
+    except SPYValueError as err:
+        assert 'pad_to_length' in str(err)
+        assert 'expected value to be greater' in str(err)
+
+    try:
+        call(pad_to_length='IamNoPad')
+    except SPYValueError as err:
+        assert 'Invalid value of `pad_to_length`' in str(err)
+        assert 'nextpow2' in str(err)
+
+    try:
+        call(pad_to_length=np.array([1000]))
+    except SPYValueError as err:
+        assert 'Invalid value of `pad_to_length`' in str(err)
+        assert 'nextpow2' in str(err)
+
+
+def run_cfg_test(call, method, positivity=True):
+
+    cfg = get_defaults(connectivity)
+
+    cfg.method = method
+    cfg.foilim = [0, 70]
+    cfg.taper = 'parzen'
+    cfg.output = 'abs'
+
+    result = call(cfg)
+
+    # check here just for finiteness and positivity
+    assert np.all(np.isfinite(result.data))
+    if positivity:
+        assert np.all(result.data[0, ...] >= -1e-10)
+
+
+def run_foi_test(call, foilim, positivity=True):
+
+    # only positive frequencies
+    assert np.min(foilim) >= 0
+    assert np.max(foilim) <= 500
+
+    # fois
+    foi1 = np.arange(foilim[0], foilim[1]) # 1Hz steps
+    foi2 = np.arange(foilim[0], foilim[1], 0.25) # 0.5Hz steps
+    foi3 = 'all'
+    fois = [foi1, foi2, foi3, None]
+
+    for foi in fois:
+        # FIXME: this works for method='granger' but not method='coh' 0.0
+
+        result = call(foi=foi, foilim=None)
+        # check here just for finiteness and positivity
+        assert np.all(np.isfinite(result.data))
+        if positivity:
+            assert np.all(result.data[0, ...] >= -1e-10)
+
+    # 2 foilims
+    foilims = [[2, 60], [7.65, 45.1234], None]
+    for foil in foilims:
+        result = call(foilim=foil, foi=None)
+        # check here just for finiteness and positivity
+        assert np.all(np.isfinite(result.data))
+        if positivity:        
+            assert np.all(result.data[0, ...] >= -1e-10)
+
+    # make sure specification of both foi and foilim triggers a
+    # Syncopy ValueError
```

Comment:
These `try`/`except` blocks could be replaced with `pytest`'s own context manager:
```python
with pytest.raises(SPYValueError) as spv:
    result = call(foi=foi, foilim=foilim)
    assert "foi/foilim" in str(spv.value)
```

Reply:
Yap.. I guess that's more a style question, I'll leave this as it is for now?!